addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -608,6 +624,57 @@ options.registry.GalleryImageList = options.registry.GalleryLayout.extend({
     // Private
     //--------------------------------------------------------------------------
 
+    /**
+     * Handles the update of the thumbnail when an image or a video is changed.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _handleImageChangedElement(ev) {
+        const index = Array.from(this.$target[0].querySelectorAll(".carousel-item")).findIndex(
+            (item) => item.classList.contains("active")
+        );
+        const targetEl = ev.target;
+        if (targetEl.classList.contains("img")) {
+            const carouselItemEl = ev.currentTarget.querySelector(
+                ".carousel:first-of-type li[data-bs-target]:nth-child(" + (index + 1) + ")"
+            );
+            if (carouselItemEl) {
+                carouselItemEl.style.backgroundImage = "url(" + targetEl.dataset.originalSrc + ")";
+            }
+            return;
+        }
+
+        const iframeEl = targetEl.querySelector("iframe");
+        if (iframeEl) {
+            const thumbnailUrl = this._extractThumbnailUrl(iframeEl.src);
+            const carouselIndicatorEls = this.$target[0].querySelectorAll(
+                ".carousel li[data-bs-target]"
+            );
+
+            if (carouselIndicatorEls[index]) {
+                carouselIndicatorEls[index].style.backgroundImage = `url(${thumbnailUrl})`;
+            }
+        }
+    },
+    /**
+     * @private
+     * @param {String} src
+     *
+     * Extract thumbnail from video src.
+     */
+    _extractThumbnailUrl(src) {
+        if (src.includes("youtube.com") || src.includes("youtu.be")) {
```

Comment:
I think this should not be duplicated here, but instead rely on `web_editor/tools.py`'s `get_video_thumbnail` through a new route.